01-messaging.md

```diff
@@ -192,6 +194,21 @@ things, enables the following optimizations:
  - variable-size fields can reserve their expected size up front, rather than
    appending elements sequentially and incurring double-and-copy overhead.
 
+The main reason for unique occurrences of some TLV record type relates back the
+fact that if a decoder doesn't recognize a field, it can't be certain whether
+there should only be one, or if multiple are permitted. Asserting this would
+require some knowledge of what the field encodes or a scheme from which it can
+be inferred knowing only the type, e.g. (a silly example) only prime types can
+have multiple records. The first defeats the purpose of the proposal, and the
+latter is cumbersome in practice and would further fragment the type space
+beyond the existing even/odd rule.
+
+A minor reasons for strictly-unique types is that permitting multiple of the
+same type is less efficient on the wire due to the overhead of redundant
+type-length pairs. It is also less efficient when allocating memory for the
+decoded objects, e.g. one can't allocate the exact size of a vector upfront and
+will incur more copying overhead when appending each item individually.
+
```

Comment:
I would remove these paragraphs entirely, they clutter the spec (IMHO).

Reply:
Well, I personally find this explanation (when was given it) quire useful; it helps understanding the reasons of particular design. The spec is not that large yet, so having two more paragraphs adding clarity is good. But I am fine to remove them LN community thinks they are unnecessary.

Reply:
I'll let @cfromknecht chime in, but I think the general feeling when we discussed this during a spec meeting was to remove or condense these paragraphs.

Reply:
Done + rebased